_docs/_latest/develop/dev/gcp-api.md

```diff
@@ -0,0 +1,213 @@
+        'supported_versions': ['v1', 'beta']
+    },
+    'sqladmin': {
+        'default_version': 'v1beta4',
+        'supported_versions': ['v1beta4']
+    },
+}
+```
+
+## Create a new foo API client file
+
+Create a new foo API client file called foo.py in
+[google/cloud/forseti/common/gcp_api](https://github.com/GoogleCloudPlatform/forseti-security/blob/stable/google/cloud/forseti/common/gcp_api)
+package.
+
+## Create a FooClient class to provide the entry point to the GCP API method
+
+1. Create FooClient class in foo.py.
+1. Initialize with the allowed quota and FooRepositoryClient.
+1. Create a get_foos(), which will invoke the actual GCP API
+call.
+
+[google/cloud/forseti/common/gcp_api/cloudsql.py](https://github.com/GoogleCloudPlatform/forseti-security/blob/stable/google/cloud/forseti/common/gcp_api/cloudsql.py)
+
+```python
+class CloudsqlClient(object):
+    """CloudSQL Client."""
+
+    def __init__(self, global_configs, **kwargs):
+        """Initialize.
+
+        Args:
+            global_configs (dict): Global configurations.
+            **kwargs (dict): The kwargs.
+        """
+        max_calls, quota_period = api_helpers.get_ratelimiter_config(
+            global_configs, 'sqladmin')
+
+        self.repository = CloudSqlRepositoryClient(
+            quota_max_calls=max_calls,
+            quota_period=quota_period,
+            use_rate_limiter=kwargs.get('use_rate_limiter', True))
+
+    def get_instances(self, project_id):
+        """Gets all CloudSQL instances for a project.
+
+        Args:
+            project_id (int): The project id for a GCP project.
+
+        Returns:
+            list: A list of database Instance resource dicts for a project_id.
+            https://cloud.google.com/sql/docs/mysql/admin-api/v1beta4/instances
+        """
+
+        paged_results = self.repository.instances.list(project_id)
+        flattened_results = api_helpers.flatten_list_results(
+            paged_results, 'items')
+        LOGGER.debug('Getting all the cloudsql instances of a project,'
+                     ' project_id = %s, flattened_results = %s',
+                      project_id, flattened_results)
+        return flattened_results
+```
+
+## Create a FooRepositoryClient class, and define the property method
+
+1. Create a FooRepositoryClient class in foo.py.
+1. Make it inherit the `BaseRepositoryClient`.
+1. Initialize it with quota parameters.
+1. Define the property method named after the bar resource we want to interact
+with, e.g. bar(). This property method will initialize a \_FooBarRepository
+class, which will install the base GCP API functionalities (building requests,
+authentication, and API methods).
+
+[google/cloud/forseti/common/gcp_api/cloudsql.py](https://github.com/GoogleCloudPlatform/forseti-security/blob/stable/google/cloud/forseti/common/gcp_api/cloudsql.py)
+
+In this example, bar is ```instances```.
+
+```python
+class CloudSqlRepositoryClient(_base_repository.BaseRepositoryClient):
```

Comment:
I think it might make more sense to reverse the order here, since they have dependencies between each others.
e.g. 
1. create the repository to make the API call.
2. expose the repository in the repository client as property.
3. create a client that contains an instance of the repository client and create the corresponding methods to use the repos client.